python/ray/autoscaler/kubernetes/defaults.yaml

```diff
@@ -207,45 +163,51 @@ worker_nodes:
         - name: dshm
           emptyDir:
-              medium: Memory
-
+            medium: Memory
         containers:
         - name: ray-node
           imagePullPolicy: Always
-          # You are free (and encouraged) to use your own container image,
-          # but it should have the following installed:
-          #   - rsync (used for `ray rsync` commands and file mounts)
           image: rayproject/ray:nightly
           # Do not change this command - it keeps the pod alive until it is
           # explicitly killed.
           command: ["/bin/bash", "-c", "--"]
-          args: ["trap : TERM INT; sleep infinity & wait;"]
+          args: ['trap : TERM INT; sleep infinity & wait;']
           ports:
-              - containerPort: 12345 # Ray internal communication.
-              - containerPort: 12346 # Ray internal communication.
+          - containerPort: 6379  # Redis port
+          - containerPort: 10001  # Used by Ray Client
+          - containerPort: 8265  # Used by Ray Dashboard
 
           # This volume allocates shared memory for Ray to use for its plasma
           # object store. If you do not provide this, Ray will fall back to
           # /tmp which cause slowdowns if is not a shared memory volume.
           volumeMounts:
-              - mountPath: /dev/shm
-                name: dshm
+          - mountPath: /dev/shm
+            name: dshm
           resources:
-              requests:
-                  cpu: 1000m
-                  memory: 512Mi
-              limits:
-                  # This memory limit will be detected by ray and split into
-                  # 30% for plasma, and 70% for workers.
-                  memory: 2Gi
-          env:
-              # This is used in the head_start_ray_commands below so that
-              # Ray can spawn the correct number of processes. Omitting this
-              # may lead to degraded performance.
-              - name: MY_CPU_REQUEST
-                valueFrom:
-                    resourceFieldRef:
-                        resource: requests.cpu
+            requests:
+              cpu: 1000m
+              memory: 512Mi
+            limits:
+              # The maximum memory that this pod is allowed to use. The
+              # limit will be detected by ray and split to use 10% for
+              # redis, 30% for the shared memory object store, and the
+              # rest for application memory. If this limit is not set and
+              # the object store size is not set manually, ray will
+              # allocate a very large object store in each pod that may
+              # cause problems for other pods.
+              memory: 512Mi
+
+
+# Command to start ray on the head node. You don't need to change this.
+# Note dashboard-host is set to 0.0.0.0 so that kubernetes can port forward.
+head_start_ray_commands:
+    - ray stop
+    - ulimit -n 65536; ray start --head --autoscaling-config=~/ray_bootstrap_config.yaml --dashboard-host 0.0.0.0
```

Comment:
is the --dashboard-host necessary?

Reply:
do we need "--autoscaling-config=~/ray_bootstrap_config.yaml"?

Reply:
You need the autoscaling config when using the cluster launcher.

You need the dashboard host.
else it'll use 127.0.0.1 and you can't portforward that.